### PR TITLE
Cleaning

### PR DESCRIPTION
**Pull request by BielinskiLukasz**
## Overview

Hello. I did some small cleaning in your repositories (simple and small changes) in Java code. Maybe you can use one or more. Have a nice day!
List of changes:
- add final keywords
- remove unnecessary keywords and semicolons
- simplify return statements
- create negate methods
- replace Optional = null by Optional.empty()
- simplify:
    - list initialization
    - create String from bytes read from file as Files.readString(xmlFile)
    - replace !Optional.isPresent() by Optional.isEmpty()
    - replace String.valueOf(c) by c in String creation // char c

---

I hereby agree to the terms of the [JUnit Contributor License Agreement](https://github.com/junit-team/junit5/blob/002a0052926ddee57cf90580fa49bc37e5a72427/CONTRIBUTING.md#junit-contributor-license-agreement).

---

### Definition of Done

- [ ] There are no TODOs left in the code
- [ ] Method [preconditions](http://junit.org/junit5/docs/snapshot/api/org/junit/platform/commons/util/Preconditions.html) are checked and documented in the method's Javadoc
- [ ] [Coding conventions](https://github.com/junit-team/junit5/blob/master/CONTRIBUTING.md#coding-conventions) (e.g. for logging) have been followed
- [ ] Change is covered by [automated tests](https://github.com/junit-team/junit5/blob/master/CONTRIBUTING.md#tests) including corner cases, errors, and exception handling
- [ ] Public API has [Javadoc](https://github.com/junit-team/junit5/blob/master/CONTRIBUTING.md#javadoc) and [`@API` annotations](https://apiguardian-team.github.io/apiguardian/docs/current/api/org/apiguardian/api/API.html)
- [ ] Change is documented in the [User Guide](http://junit.org/junit5/docs/snapshot/user-guide/) and [Release Notes](http://junit.org/junit5/docs/snapshot/user-guide/#release-notes)
- [ ] All [continuous integration builds](https://github.com/junit-team/junit5#continuous-integration-builds) pass

### CODE REVIEW

File: documentation/src/test/java/example/testinterface/TestLifecycleLogger.java [hunk 1]
```diff
@@ -25,7 +25,7 @@
 @TestInstance(Lifecycle.PER_CLASS)
 interface TestLifecycleLogger {
 
-	static final Logger logger = Logger.getLogger(TestLifecycleLogger.class.getName());
+	Logger logger = Logger.getLogger(TestLifecycleLogger.class.getName());
```
Comment (toniedzwiedz) [inline documentation/src/test/java/example/testinterface/TestLifecycleLogger.java]: I wouldn't say these keywords were superfluous. There's no reason to make a logger per instance and I don't see why anyone would want to reassign the logger.

File: junit-jupiter-engine/src/main/java/org/junit/jupiter/engine/extension/TestInfoParameterResolver.java [hunk 2]
```diff
@@ -85,7 +85,7 @@ public String toString() {
 		}
 
 		private static Object nullSafeGet(Optional<?> optional) {
-			return optional != null ? optional.orElse(null) : null;
+			return optional.isPresent() ? optional.orElse(null) : null;
```
Comment (toniedzwiedz) [inline junit-jupiter-engine/src/main/java/org/junit/jupiter/engine/extension/TestInfoParameterResolver.java]: This change breaks null-safety. `null.isPresent()` will crash and burn.

Comment (BielinskiLukasz) [reply junit-jupiter-engine/src/main/java/org/junit/jupiter/engine/extension/TestInfoParameterResolver.java]: Thanks for the review, you are right, I didn't think about that situation

File: documentation/src/test/java/example/testinterface/TestLifecycleLogger.java (hunk 1 above)
Comment (BielinskiLukasz) [reply documentation/src/test/java/example/testinterface/TestLifecycleLogger.java]: Of course, you are right :) And also I have. Fields in an interface are by default public, static and final.

**Comment by BielinskiLukasz** [comment]
I made changes also in unit test classes so it caused problems with checks. I will try to clean this tomorrow (revert changes in unit tests).

**Comment by BielinskiLukasz** [comment]
I must revert many of own changes. Maybe I will create a new pull request without changes in unit tests (and NullpointerException found by @toniedzwiedz). If that's better idea please close that request or write a comment. Thanks a lot! :)

**Comment by sbrannen** [comment]
> I must revert many of own changes. Maybe I will create a new pull request without changes in unit tests (and NullpointerException found by @toniedzwiedz). If that's better idea please close that request or write a comment. Thanks a lot! :)

OK. I'll close this PR now.

Please submit new PRs for the individual categories of changes you propose and take note of the tips I provide below.

That will make it easier for the team to digest.

We typically don't care about a field or a local variable in a test class being `final`. So we will likely choose not to accept such proposed changes.

The same holds for stylistic decisions like the presence of a trailing semicolon for enum constants. We've done that consistently and do not plan to change that.

**Pull request closed without merging**